modules/lru_cache.py

```diff
@@ -0,0 +1,98 @@
+from threading import RLock
+
+
+class _Node:
+    def __init__(self, key, value):
+        self.key = key
+        self.value = value
+        self.next = None
+        self.prev = None
+
+
+class LruCache:
+    def __init__(self):
+        self._cache = {}
+        self._head = _Node(0, 0)
+        self._tail = _Node(0, 0)
+        self._head.next = self._tail
+        self._tail.prev = self._head
+        self._lock = RLock()
+
+    def pop(self):
+        """
+        remove a key from cache and return the removed key
+        Returns:
+
+        """
+        with self._lock:
+            if len(self._cache) > 0:
+                node = self._head.next
+                self._link_remove(node)
+                del self._cache[node.key]
+                return node.key
```

Comment:
For convinience, should you also return the value of the node?

Reply:
done